src/App.tsx

```diff
@@ -8,55 +8,137 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getUserPosts, getUsers } from './api/users';
+import { Post } from './types/Post';
+import { User } from './types/User';
+import { ErrorMessage } from './types/Error';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+  const [isLoaling, setIsLoaling] = useState(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [errorType, setErrorType] = useState<ErrorMessage>(ErrorMessage.None);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [sidebarIsOpen, setSidebarIsOpen] = useState(false);
+
+  useEffect(() => {
+    getUsers()
+      .then(usersFromServer => setUsers(usersFromServer));
+  }, []);
+
+  const hendleLoadPosts = async (userId: number) => {
+    setSidebarIsOpen(false);
+    setSelectedPost(null);
+    setUserPosts([]);
+    setSelectedUserId(userId);
+    setIsLoaling(true);
+    try {
+      const postFromServer = await getUserPosts(userId);
+
+      setErrorType(ErrorMessage.None);
+
+      setUserPosts(postFromServer);
+
+      if (!postFromServer.length) {
+        setErrorType(ErrorMessage.UserPosts);
+      }
+    } catch {
+      setErrorType(ErrorMessage.UserPosts);
+    } finally {
+      setIsLoaling(false);
+    }
+  };
+
+  const heldleOpenSidebar = (post: Post) => {
```

Comment:
Fix typo, what does it mean `heldle`? Maybe you need `handle`?
```suggestion
  const heldleOpenSidebar = (post: Post) => {
```